src/app/src/main/kotlin/org/imperial/mrc/hint/exceptions/ExceptionHandler.kt

```diff
@@ -34,10 +35,18 @@ class HintExceptionHandler(private val errorCodeGenerator: ErrorCodeGenerator,
         return unexpectedError(status, request, e.message)
     }
 
-    @ExceptionHandler(PSQLException::class)
-    fun handlePSQLException(e: PSQLException, request: WebRequest): ResponseEntity<Any> {
-        logger.error(e.message)
-        // for security reasons we should not return arbitrary db errors to the frontend
+    @ExceptionHandler(Exception::class)
+    fun handleArbitraryException(e: Exception, request: WebRequest): ResponseEntity<Any> {
+        val error = if (e is UndeclaredThrowableException) {
```

Comment:
When you handle *all* exceptions, you start catching these - they happen when a method isn't marked with the `@Throws` annotation, and wrap the actual error that was thrown.

Reply:
Could this be unit tested?